src/tablet/tablet_impl.cc

```diff
@@ -208,13 +224,39 @@ bool TabletImpl::Init(const std::string& zk_cluster, const std::string& zk_path,
         return false;
     }
 
-    if (!CreateMultiDir(mode_root_paths_)) {
-        PDLOG(WARNING, "fail to create db root path %s", FLAGS_db_root_path.c_str());
+    if (FLAGS_db_root_path != "" && !CreateMultiDir(mode_root_paths_[::openmldb::common::kMemory])) {
+        PDLOG(WARNING, "fail to create db root path %s",
+              FLAGS_db_root_path.c_str());
+        return false;
+    }
+
+    if (FLAGS_ssd_root_path != "" && !CreateMultiDir(mode_root_paths_[::openmldb::common::kSSD])) {
+        PDLOG(WARNING, "fail to create ssd root path %s",
+              FLAGS_ssd_root_path.c_str());
+        return false;
+    }
+
+    if (FLAGS_hdd_root_path != "" && !CreateMultiDir(mode_root_paths_[::openmldb::common::kHDD])) {
+        PDLOG(WARNING, "fail to create hdd root path %s",
+              FLAGS_hdd_root_path.c_str());
+        return false;
+    }
+
+    if (FLAGS_recycle_bin_root_path != "" && !CreateMultiDir(mode_recycle_root_paths_[::openmldb::common::kMemory])) {
```

Comment:
if `FLAGS_recycle_bin_enabled` is true and `FLAGS_db_root_path` is not empty,
if FLAGS_recycle_bin_root_path == "", return false;

same for `hdd` and `ssd`